### PR TITLE
Categorize certain aws-route53 errors as configuration problems

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement
/platform aws

**What this PR does / why we need it**:
Categorizes certain aws-route53 errors as configuration problems as suggested in https://github.com/gardener/gardener/issues/4294.

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardener/issues/4294

**Special notes for your reviewer**:

**Release note**:

```other operator
Failures to reconcile `DNSRecords` due to a missing hosted zone or a DNS name not matching the zone name are now properly categorized as `ERR_CONFIGURATION_PROBLEM`.
```

### CODE REVIEW

**Review by ialidzhikov — DISMISSED** [review]
/lgtm

File: pkg/aws/client/client_route53.go [hunk 1]
```diff
@@ -315,3 +316,19 @@ func isValuesDoNotMatchError(err error) bool {
 	}
 	return false
 }
+
```
Comment (dkistner) [inline pkg/aws/client/client_route53.go:319]: Please add doc string

File: pkg/aws/client/client_route53.go [hunk 2]
```diff
@@ -315,3 +316,19 @@ func isValuesDoNotMatchError(err error) bool {
...
+func IsNoSuchHostedZoneError(err error) bool {
+	if aerr, ok := err.(awserr.Error); ok && aerr.Code() == route53.ErrCodeNoSuchHostedZone {
+		return true
+	}
+	return false
+}
+
+var notPermittedInZoneRegex = regexp.MustCompile(`RRSet with DNS name [^\ ]+ is not permitted in zone [^\ ]+`)
+
+func IsNotPermittedInZoneError(err error) bool {
```
Comment (dkistner) [inline pkg/aws/client/client_route53.go:329]: Please add doc string

**Review by dkistner — COMMENTED** [review]
Looks good to me. Just two missing doc strings.

**Review by rfranzke — DISMISSED** [review]
/lgtm apart from the nits mentioned by @dkistner 

I guess we still
/hold
as there is the ongoing release validation? cc @ialidzhikov

**Comment by ialidzhikov** [comment]
I think it should be fine to include this one as it affects only the dnsrecord controller which is still not heavily used.

/unhold

File: pkg/aws/client/client_route53.go (hunk 1 above)
Comment (stoyanr) [reply pkg/aws/client/client_route53.go:319]: Done

File: pkg/aws/client/client_route53.go (hunk 2 above)
Comment (stoyanr) [reply pkg/aws/client/client_route53.go:331]: Done

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**